charts/keycloakx/templates/statefulset.yaml

```diff
@@ -156,10 +156,12 @@ spec:
           livenessProbe:
             {{- tpl . $ | nindent 12 }}
           {{- end }}
+          {{- if .Values.metrics.enabled }}
```

Comment:
Does this line forces enabling metrics in order to have any readinessProbe?
If yes, IMO it would reduce flexibility of defining a custom command based probe.

For me info in documentation is sufficient for those who want to use Keycloak health checks :)

Furthermore, the endpoint is still active even when metrics are disabled:

1) metrics enabled:
```
bash-4.4$ curl -k https://127.0.0.1:8443/health/ready
{
    "status": "UP",
    "checks": [
        {
            "name": "Keycloak database connections health check",
            "status": "UP"
        }
    ]
}
```
2) metrics disabled:
curl -k https://127.0.0.1:8443/health/ready
{
    "status": "UP",
    "checks": [
    ]
}

It's not clear for me whether I really need the `Database` check enabled for readiness checking:
https://www.keycloak.org/server/health

And since now we have the `Database` check maybe the `dbchecker` functionality is not needed anymore and could be deprecated.

Reply:
@Footur you marked this discussion as resolved. @tomasz-baran pointed out a valid issue here i think - we reduce the flexibility here; maybe we should just drop the line here, but keep the information in the readme. This way one can configure the probe as needed (for whatever workaround one might want to implement).

Reply:
Additionally, i know, i requested the change initially. But with the issue @tomasz-baran pointed out, my "old" request might be not valid..

Reply:
@grieshaber Thanks for the comments. I've checked Keycloak code and you are right. The `ready` check is defined in `KeycloakReadyHealthCheck` class which extends `DataSourceHealthCheck`. There is also PR with design doc with 
[a comment](https://github.com/keycloak/keycloak-community/pull/55/files#r528560159) on this topic.

I was misleading by the Keycloak Operator code. However, it looks like Keycloak project wants to not force enabling metrics in order to use the `/health/ready` endpoint - based on their test code:
```
    @Test
    @Launch({ "start-dev", "--health-enabled=true" })
    void testHealthEndpoint() {
        when().get("/health").then()
                .statusCode(200);
        when().get("/health/live").then()
                .statusCode(200);
        when().get("/health/ready").then()
                .statusCode(200);
        // Metrics should not be enabled
        when().get("/metrics").then()
                .statusCode(404);
    }
```

Reply:
Ok - i'd say, let's stick to the current implementation, as this implies the most valid readiness check for the software the chart chips. If another use-case arises, we can adapt again to that use-case.